pkg/apis/navigator/validation/cassandra.go

```diff
@@ -33,15 +33,24 @@ func ValidateCassandraClusterUpdate(old, new *navigator.CassandraCluster) field.
 
 		for _, oldNp := range old.Spec.NodePools {
 			if newNp.Name == oldNp.Name {
-				if newNp.Rack != oldNp.Rack {
-					allErrs = append(allErrs, field.Forbidden(idxPath.Child("rack"), "cannot modify rack"))
+				if !reflect.DeepEqual(newNp.Persistence, oldNp.Persistence) {
+					if oldNp.Persistence.Enabled {
+						allErrs = append(allErrs, field.Forbidden(idxPath.Child("persistence"), "cannot modify persistence configuration once enabled"))
+					}
 				}
-				if newNp.Datacenter != oldNp.Datacenter {
-					allErrs = append(allErrs, field.Forbidden(idxPath.Child("datacenter"), "cannot modify datacenter"))
-				}
-				if !reflect.DeepEqual(newNp.NodeSelector, oldNp.NodeSelector) {
-					allErrs = append(allErrs, field.Forbidden(idxPath.Child("nodeSelector"), "cannot modify nodeSelector"))
+
+				restoreReplicas := newNp.Replicas
+				newNp.Replicas = oldNp.Replicas
+
+				restorePersistence := newNp.Persistence
+				newNp.Persistence = oldNp.Persistence
```

Comment:
:grey_question: I wonder why it's necessary to restore the original values? I see that's what they do in the StatefulSet validation, but I thought that validation didn't have opportunity to mutate the supplied objects. And I guess you can't mutate the old object either, otherwise why not do this the other way round? Assign the new value to the old object and then compare?

Reply:
They do that in the statefulset validation because they do additional validation afterwards (`apivalidation.ValidateNonnegativeField`). I added it in case we want to do the same in the future, but can remove if it's extra noise.